pkg/apis/flux/v1alpha1/validation/fluxconfig.go

```diff
@@ -125,6 +126,9 @@ func ValidateAdditionalSecretResources(additionalResources []fluxv1alpha1.Additi
 	}
 
 	for i, r := range additionalResources {
+		if ptr.Deref(r.TargetName, "") != "" && len(validation.IsDNS1123Subdomain(*r.TargetName)) > 0 {
```

Comment:
Can you switch to `NameIsDNSSubdomain`?
This is what the k8s secret validation uses: https://github.com/kubernetes/kubernetes/blob/dbc2b0a5c7acc349ea71a14e49913661eaf708d2/pkg/apis/core/validation/validation.go#L279-L282

Reply:
which is just a wrapper around this function with an additional argument...
https://github.com/kubernetes/apimachinery/blob/v0.30.3/pkg/api/validation/generic.go#L37

Reply:
Ah, right.